docs/example.md

```diff
@@ -0,0 +1,509 @@
+      load_result = load_ensemble(trim(input_file))
+      if (load_result%error_code /= SW_SUCCESS) then
+        print *, "isotherms_f90: ", trim(load_result%error_message)
+        stop
+      end if
+
+      ensemble = load_result%ensemble
+      print *, "isotherms_f90: found ", ensemble%size, " ensemble members."
+
+      ! Iterate over all members of the ensemble.
+      do while (ensemble%next(input, output))
+        ! Fetch inputs.
+        V = input%get("V") ! gas (molar) volume [m3]
+        T = input%get("T") ! gas temperature [K]
+
+        ! Compute p(V, T).
+        p = R * T / V
+
+        ! Stash the computed pressure in the member's output.
+        call output%set("p", p);
+      end do
+
+      ! Write out a Python module.
+      output_file = "ideal_gas_isotherms_f90.py"
+      print *, "isotherms_f90: Writing data to ", output_file, "..."
+      call ensemble%write(output_file)
+
+      ! Clean up.
+      call ensemble%free();
+    end program
+    ```
+
+Click through the different tabs for the C, C++, and Fortran versions of the
+program. You can see that they're very similar. Here's what's going on:
+
+1. An ensemble is loaded with a `load_ensemble` function. If the attempt to
+   load the ensemble fails, the program halts.
+2. The members of the ensemble are processed one by one. In the C and Fortran
+   programs, this happens in a `while`/`do while` loop, in which a call to
+   `sw_ensemble_next` or `ensemble%next` associates the `input` and `output`
+   variables with a specific ensemble member. In C++, this loop is hidden inside
+   a call to `ensemble->process`, which accepts a lambda function whose
+   arguments are the properly associated `input` and `output` variables for each
+   member. Inside this loop:
+    1. the gas volume $V$ and temperature $T$ are extracted from the `input`
+       variable for the current ensemble member
+    2. the gas pressure $p$ is computed from $V$ and $T$
+    3. the computed pressure $p$ is stored in the `output` variable for the
+       current ensemble member
+3. After every ensemble member has been processed in Step 2, a Python module
+   containing all input and output data for the ensemble is written.
+4. The ensemble is deleted with an appropriate function call or command.
+
+For a more detailed explanation of any of these steps, see the [API](api.md)
+section.
+
+If you like, copy and paste any version of the program into a text editor, save
+it, and build it, linking it against `libskywalker.a` or `libskywalker_f90.a`
+as needed. Then run it in a directory containing `ideal_gas_isotherms.yaml`,
+and check to see that an appropriate `.py` file appears.
+
+Here's a plot of the resulting isotherms:
+
+![Ideal gas law isotherms](images/ideal_gas_isotherms_c.png)
+
+If you want to generate this plot for yourself, run the
+[plot_isotherms.py](https://github.com/eagles-project/skywalker/tree/main/examples/plot_isotherms.py)
```

Comment:
Is the skywalker/tree/main/examples directory supposed to currently exist in the skywalker repository?

Reply:
Thanks for going through the docs, James! That directory will only exist after the PR gets merged, so there are a few dead links in there at the moment. I didn't want to use the branch path, which is https://github.com/eagles-project/skywalker/tree/docs/examples/plot_isotherms.py, because we'd have to change it anyway once it's merged.